src/foam/nanos/auth/ChangePasswordView.js

```diff
@@ -205,33 +201,13 @@ foam.CLASS({
           return;
         }
 
-        // validate new password
+        // check if new password entered
         if ( ! this.newPassword ) {
           this.add(this.NotificationMessage.create({ message: this.emptyPassword, type: 'error' }));
           return;
         }
 
-        if ( this.newPassword.includes(' ') ) {
-          this.add(this.NotificationMessage.create({ message: this.noSpaces, type: 'error' }));
-          return;
-        }
-
-        if ( this.newPassword.length < 7 || this.newPassword.length > 32 ) {
-          this.add(this.NotificationMessage.create({ message: this.invalidLength, type: 'error' }));
-          return;
-        }
-
-        if ( ! /\d/g.test(this.newPassword) ) {
-          this.add(self.NotificationMessage.create({ message: this.noNumbers, type: 'error' }));
-          return;
-        }
-
-        if ( /[^a-zA-Z0-9]/.test(this.newPassword) ) {
-          this.add(self.NotificationMessage.create({ message: this.noSpecial, type: 'error' }));
-          return;
-        }
-
-        // check if confirmation entered
+        // check if new password confirmation entered
```

Comment:
I think we should get the message from the password validator (via Exception?). Internationalization is going to be a pain, but we can worry about that later.

Reply:
Just to confirm that I understand - are you talking about a particular one of these messages or all of them?

Currently, the authService only receives the old password and the new password. 
The check that the new password and the confirmation match is currently only on the front end. 

To move all messages to the service would require that we pass everything to the service, and perform all the checks there. 

I myself think it's probably cleaner to move all messages and logic to the service.

@kgrgreer

Reply:
Services - yes all password validation/reset/change and subsequent messaging should reside service side.  Consider and LDAPAuthService - this validation would be occurring in the LDAP server.

Reply:
Noted.